e2e/testcases/invalid_git_branch_test.go

```diff
@@ -22,13 +22,15 @@ import (
 	"kpt.dev/configsync/e2e/nomostest/ntopts"
 	nomostesting "kpt.dev/configsync/e2e/nomostest/testing"
 	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/api/configsync/v1beta1"
 	"kpt.dev/configsync/pkg/core"
 	"kpt.dev/configsync/pkg/status"
 	"kpt.dev/configsync/pkg/testing/fake"
 )
 
 func TestInvalidRootSyncBranchStatus(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipMonoRepo)
+	nt.T.Cleanup(resetToMain(nt))
```

Comment:
Line 49 already updates RootSync to sync from the main branch, no need to reset again.
If we want to guarantee the reset, we can move line 49 to 58 to the cleanup block.

Reply:
Moved the reset and metrics validation into a cleanup block.